server/src/test/java/org/elasticsearch/cluster/coordination/CoordinatorTests.java

```diff
@@ -158,6 +161,48 @@ public void testDoesNotElectNonMasterNode() {
         }
     }
 
+    public void testNoElectedLeaderForNonWritableNodes() {
+        try (Cluster cluster = new Cluster(randomIntBetween(1, 5), true, Settings.EMPTY, () -> NodeHealthService.Status.UNHEALTHY)) {
+            cluster.runRandomly();
+            cluster.runFor(DEFAULT_STABILISATION_TIME, "allowing time for stabilization");
+            final List<ClusterNode> leaders = cluster.getAllLeaders();
+            assertThat(leaders, empty());
+        }
+    }
+
+    public void testNonWritableNodeCannotBecomeLeader() {
+        try (Cluster cluster = new Cluster(randomIntBetween(3, 5))) {
+            cluster.runRandomly();
+            cluster.stabilise();
+
+            final ClusterNode leader = cluster.getAnyLeader();
+            ClusterNode nonWritableNode = cluster.new ClusterNode(nextNodeIndex.getAndIncrement(), true, leader.nodeSettings,
+                () -> NodeHealthService.Status.UNHEALTHY);
+            cluster.clusterNodes.add(nonWritableNode);
```

Comment:
Could we instead dynamically change a node to become unhealthy and verify that it's removed from the cluster?